packages/kilt-did/README.md

```diff
@@ -3,6 +3,320 @@
+
+By design, DID signatures and Substrate signatures are decoupled, meaning that the encoded and signed DID creation operation can then be signed and submitted by a different KILT account than the DID subject. We think this opens the path for a wider range of use cases in which, for instance, a service provider might be willing to offer a DID-as-a-Service option for its customers.
+
+An example of a full DID is the following:
+
+```
+did:kilt:4rp4rcDHP71YrBNvDhcH5iRoM3YzVoQVnCZvQPwPom9bjo2e
+```
+
+Here, there is no `light:` component, which indicates that the DID is a full DID and that the keys associated with it must not be derived from the DID identifier but must be retrieved from the KILT blockchain.
+
+Beyond an authentication key, and encryption key, and external services, a full DID also supports an **attestation key**, which must be used to write CTypes and attestations on the blockchain, and a **delegation key**, which must be used to write delegations on the blockchain.
+
+### Creating and anchoring a full DID
+
+The following is an example of how to create and write a full DID on chain that specifies only an authentication key.
+
+```typescript
+import Keyring from '@polkadot/keyring'
+
+import Kilt from '@kiltprotocol/sdk-js'
+import { BlockchainUtils } from '@kiltprotocol/chain-helpers'
+import {
+  DefaultResolver,
+  DemoKeystore,
+  DidUtils,
+  SigningAlgorithms
+} from '@kiltprotocol/did'
+import { KeyRelationship } from '@kiltprotocol/types'
+
+// Initialise connection to a KILT blockchain node.
+await Kilt.init({ address: 'ws://127.0.0.1:9944' })
+
+// Generate the KILT account that will submit the DID creation tx to the KILT blockchain.
+// It must have enough funds to pay for the tx execution fees.
+const aliceKiltAccount = new Keyring({
+  type: 'ed25519',
+  // KILT has registered the ss58 prefix 38
+  ss58Format: 38,
+}).createFromUri('//Alice')
+
+// Instantiate the demo keystore.
+const keystore = new DemoKeystore()
+
+// Generate seed for the authentication key.
+const authenticationSeed = '0x123456789'
+
+// Ask the keystore to generate a new keypar to use for authentication.
+const authenticationKeyPublicDetails = await keystore.generateKeypair({
+  seed: authenticationSeed,
+  alg: SigningAlgorithms.Ed25519,
+})
+
+// Generate the DID-signed creation extrinsic.
+// The extrinsic is unsigned and contains the DID creation operation signed with the DID authentication key.
+const { submittable, did } = await DidUtils.writeDidfromPublicKeys(keystore, {
+  [KeyRelationship.authentication]: {
+    publicKey: authenticationKeyPublicDetails.publicKey,
+    type: authenticationKeyPublicDetails.alg,
+  },
+})
+// Will print `did:kilt:4sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar`.
+console.log(did)
+
+// Submit the DID creation tx to the KILT blockchain after signing it with the KILT account specified.
+await BlockchainUtils.signAndSubmitTx(submittable, aliceKiltAccount, {
+  resolveOn: BlockchainUtils.IS_IN_BLOCK,
```

Comment:
I’ve seen that much of the documentation uses `resolveOn: BlockchainUtils.IS_IN_BLOCK`, however the recommendation for the Sporran team was to use `resolveOn: BlockchainUtils.IS_FINALIZED`. Should we also recommend that to other developers?

Reply:
IS_IN_BLOCK would be prone to consensus errors afaik (correct me if im wrong)
For the extension or other stuff where you want to ensure that the tx is finalized, use IS_FINALIZED
Tho ofc IS_FINALIZED takes 2 blocks longer

Reply:
I think this is made clear (or should be made clear) in the doc somewhere else. And I also think that for demo purposes such as a README for DIDs, using `IS_IN_BLOCK` would allow for faster testing.

Reply:
Good documentation does not require you to read the entirety of it to make a simple thing right. Currently, this example is a trap. I see the point about quicker testing, but it should rather be done the other way: correct code in the example, comment about speeding up somewhere. Or declaring above: `const resolveOn = process.env.NODE_ENV === 'production' ? BlockchainUtils.IS_FINALIZED : BlockchainUtils.IS_IN_BLOCK`

Reply:
I have a bit about it in my tx submission doc ready for review here : https://github.com/KILTprotocol/ticket/issues/1191

This doc is for Majella and her documentation efforts.